File: frontend/safecolony_app/lib/features/community_services/models/community_service.dart
class CommunityServiceEntry {
  final int id;
  final String name;
  final String category;
  final String phone;
  final String workDescription;
  final String? notes;
  final bool isActive;
  final String? createdByName;
  final String? updatedByName;
  final DateTime createdAt;
  final DateTime updatedAt;

  const CommunityServiceEntry({
    required this.id,
    required this.name,
    required this.category,
    required this.phone,
    required this.workDescription,
    this.notes,
    required this.isActive,
    this.createdByName,
    this.updatedByName,
    required this.createdAt,
    required this.updatedAt,
  });

  factory CommunityServiceEntry.fromJson(Map<String, dynamic> json) {
    return CommunityServiceEntry(
      id: (json['id'] as num).toInt(),
      name: json['name']?.toString() ?? '',
      category: json['category']?.toString() ?? '',
      phone: json['phone']?.toString() ?? '',
      workDescription: json['work_description']?.toString() ?? '',
      notes: json['notes']?.toString(),
      isActive: json['is_active'] == true,
      createdByName: json['created_by_name']?.toString(),
      updatedByName: json['updated_by_name']?.toString(),
      createdAt: DateTime.parse(json['created_at'].toString()),
      updatedAt: DateTime.parse(json['updated_at'].toString()),
    );
  }
}

File: backend/app/schemas/community_service.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CommunityServiceCreate(BaseModel):
    name: str = Field(min_length=2, max_length=120)
    category: str = Field(min_length=2, max_length=80)
    phone: str = Field(min_length=5, max_length=30)
    work_description: str = Field(min_length=2, max_length=255)
    notes: str | None = Field(default=None, max_length=1000)


class CommunityServiceUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=120)
    category: str | None = Field(default=None, min_length=2, max_length=80)
    phone: str | None = Field(default=None, min_length=5, max_length=30)
    work_description: str | None = Field(default=None, min_length=2, max_length=255)
    notes: str | None = Field(default=None, max_length=1000)
    is_active: bool | None = None


class CommunityServiceResponse(BaseModel):
    id: int
    name: str
    category: str
    phone: str
    work_description: str
    notes: str | None = None
    is_active: bool
    created_by_name: str | None = None
    updated_by_name: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
